provider/clusterrosa/hcp/datasource.go

```diff
@@ -0,0 +1,284 @@
+				Description: "The currently running version of OpenShift on the cluster, for example '4.11.0'.",
+				Computed:    true,
+			},
+			"state": schema.StringAttribute{
+				Description: "State of the cluster.",
+				Computed:    true,
+			},
+			"channel_group": schema.StringAttribute{
+				Description: deprecatedMessage,
+				Computed:    true,
+			},
+			"version": schema.StringAttribute{
+				Description: deprecatedMessage,
+				Computed:    true,
+			},
+			"upgrade_acknowledgements_for": schema.StringAttribute{
+				Description: deprecatedMessage,
+				Computed:    true,
+			},
+
+			// Deprecated Attributes:
+			// Those attributes were copied from cluster_rosa_clasic resource in order to use the same state struct.
+			// We can't change the rosa_classic struct to include Embedded Structs due to that issue: https://github.com/hashicorp/terraform-plugin-framework/issues/242
+			// If we will remove those attributes from the schema we will get a parsing error in the Read function
+			"disable_waiting_in_destroy": schema.BoolAttribute{
+				Description: deprecatedMessage,
+				Computed:    true,
+			},
+			"destroy_timeout": schema.Int64Attribute{
+				Description: deprecatedMessage,
+				Computed:    true,
+			},
+			"wait_for_create_complete": schema.BoolAttribute{
+				Description: deprecatedMessage,
+				Computed:    true,
+			},
+			// "autoscaling_enabled": schema.BoolAttribute{
+			// 	Description: deprecatedMessage,
+			// 	Computed:    true,
+			// },
+			"replicas": schema.Int64Attribute{
+				Description: deprecatedMessage,
+				Computed:    true,
+			},
+			"compute_machine_type": schema.StringAttribute{
+				Description: deprecatedMessage,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func (r *ClusterRosaHcpDatasource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	connection, ok := req.ProviderData.(*sdk.Connection)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *sdk.Connaction, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.clusterCollection = connection.ClustersMgmt().V1().Clusters()
+	r.versionCollection = connection.ClustersMgmt().V1().Versions()
+	r.clusterWait = common.NewClusterWait(r.clusterCollection)
+}
+
+func (r *ClusterRosaHcpDatasource) Read(ctx context.Context, request datasource.ReadRequest,
+	response *datasource.ReadResponse) {
+	tflog.Debug(ctx, "begin Read()")
+	// Get the current state:
+	state := &ClusterRosaHcpState{}
+	diags := request.Config.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	// Find the cluster:
+	get, err := r.clusterCollection.Cluster(state.ID.ValueString()).Get().SendContext(ctx)
+	if err != nil && get.Status() == http.StatusNotFound {
```

Comment:
I think this can be simplified to:
```suggestion
	if get.Status() == http.StatusNotFound {
```

Reply:
hmm, I agree it could be like that. I don't know if there's a reason for it as it's the pattern followed for all not found errors at the moment

Reply:
Yeah. Leaving it is harmless although useless (unless I'm missing something).